### PR TITLE
PYMT-897 Environment variable to dictate whether middleware is applied to Lumen

### PR DESCRIPTION
**Pull request by VeryStrongFingers**
This PR will obsolete the requirement for the consuming application to integrate middleware (if desired)
```
$app->middleware([
    AuditMiddleware::class
]);
```
^ such code is not required in `bootstrap.php`, but will not be a breaking change if not changed.
The service provider will include the middleware as part of the `boot()` method.

New environment variable introduced `AUDITING_MIDDLEWARE`, defaults to `true`

### CODE REVIEW

File: tests/Bridge/Laravel/Providers/LoyaltyCorpAuditingProviderTest.php [hunk 1]
```diff
@@ -38,10 +41,56 @@
...
+    public function testMiddlewareAppliedWhenEnabled(): void
+    {
+        /**
+         * Illuminate's packages have odd type hints
+         *
+         * @var \Tests\LoyaltyCorp\Auditing\Stubs\Vendor\Illuminate\ApplicationStub $application
+         * @var \Illuminate\Contracts\Foundation\Application $application
+         */
+        $application = new ApplicationStub();
+        \putenv('AUDITING_MIDDLEWARE=true');
```
Comment (VeryStrongFingers) [inline tests/Bridge/Laravel/Providers/LoyaltyCorpAuditingProviderTest.php]: @sjdaws with modifying environment variables for tests, should I restore to the default value afterwards?

Comment (sjdaws) [reply tests/Bridge/Laravel/Providers/LoyaltyCorpAuditingProviderTest.php]: In a test always use `setEnv` and `getEnv` so they're automatically cleaned up

Comment (sjdaws) [reply tests/Bridge/Laravel/Providers/LoyaltyCorpAuditingProviderTest.php]: You should also add this value to phpunit.xml

**Review by sjdaws — CHANGES_REQUESTED** [review]

**Comment by merk** [comment]
I dont like this approach. The application should be the thing that defines how the middleware runs.

We should have the environment checking occur in bootstrap/app.php and if the auditing middleware is disabled it should just not be registered.

**Comment by VeryStrongFingers** [comment]
> I dont like this approach. The application should be the thing that defines how the middleware runs.
> 
> We should have the environment checking occur in bootstrap/app.php and if the auditing middleware is disabled it should just not be registered.

@merk The application is in complete control of whether the middleware runs, via the environment variable and the entire package is isolated within the package (bridge) this way.
If your application is registering the service provider, it seemed logical to presume they'll want the integrate the middleware as well.

Would a dedicated service provider that's sole purpose is to register the middleware into the application stack be sufficient? or should I just entirely move this logic into subscriptions itself

**Comment by VeryStrongFingers** [comment]
Closing to add this level of control on the consuming application instead

**Pull request closed without merging**